src/net/parsers/texttracks/webvtt.js

```diff
@@ -53,82 +53,79 @@ export default function parseWebVTT(text) {
   return cuesArray;
 }
 
-/**
- * Finds from supposed first line of element 
- * (WebVTT is line based format) the type of it. It can be:
- * - note: A comment
- * - style: A style indication, about all cues, or specific cues.
- * - cue: The subtitle content.
- * - region: Informations about wrapping area of subtitle
- * @param {string} text 
- */
-function findElementType(text) {
-  if (text.match(/^NOTE.*?/g)) {
-    return "note";
-  }
-  else if (text.match(/^STYLE.*?/g)) {
-    return "style";
-  }
-  else if (text.match(/^REGION.*?/g)) {
-    return "style";
-  }
-  else if(text.length !== 0){
-    return "cue";
-  }
+function isStartOfStyleBlock(text) {
+  return text.match(/^STYLE.*?/g);
+}
+
+function isStartOfNoteBlock(text) {
+  return text.match(/^NOTE.*?/g);
+}
+
+function isStartOfRegionBlock(text) {
+  return text.match(/^REGION.*?/g);
+}
+
+function isStartOfCueBlock(text) {
+  return (!isStartOfNoteBlock(text) &&
+   !isStartOfStyleBlock(text) &&
+   !isStartOfRegionBlock(text));
 }
 
 /**
  * 
  * Parse style element from WebVTT.
  * @param {number} index 
- * @param {Array<string>} linified 
- * @param {Array<Object>} styleElements 
+ * @param {Array.<string>} linified 
  */
-function parseStyle(index, linified, styleElements) {
+function parseStyleBlock(styleBlock) {
 
+  const styleElements = [];
+  let index = 1;
   const className = [];
-  const cueClassLine = linified[index].match(/cue\(\.{0,1}(.*?)\)/);
+  const cueClassLine = styleBlock[index].match(/cue\(\.{0,1}(.*?)\)/);
   className.push(cueClassLine ? cueClassLine[1] : "cue");
 
   index++;
 
-  while(linified[index].match(/cue\(\.{0,1}(.*?)\)/)) {
-    const cueClassLine = linified[index].match(/cue\(\.{0,1}(.*?)\)/);
+  while( styleBlock[index].match(/cue\(\.{0,1}(.*?)\)/)) {
+    const cueClassLine =  styleBlock[index].match(/cue\(\.{0,1}(.*?)\)/);
     className.push(cueClassLine ? cueClassLine[1] : "cue");
     index++;
```

Comment:
Actually, they can. Each time parser encouters a STYLE bloc, it will check for every class which are concerned.

```
video::cue(c),
video::cue(i),
video::cue(b),
video::cue(u),
video::cue(ruby),
video::cue(rt),
video::cue(v),
video::cue(lang) {
  color: yellow;
}
```